prowler/providers/azure/azure_provider.py

```diff
@@ -244,14 +380,86 @@ def setup_session(
 
         return credentials
 
-    def __check_service_principal_creds_env_vars__(self):
+    @staticmethod
+    def test_connection(
+        az_cli_auth,
+        sp_env_auth,
+        browser_auth,
+        managed_identity_auth,
+        tenant_id,
+        region,
```

Comment:
Could you set default values for the function arguments?

Reply:
Done